test/debugger.js

```diff
@@ -0,0 +1,266 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+/*!
+ * @module debug/debugger
+ */
+
+var pjson = require('../package.json');
+var assert = require('assert');
+var common = require('@google-cloud/common');
+var qs = require('querystring');
+var util = require('util');
+var Debug = require('../src');
+
+/** @const {string} Cloud Debug API endpoint */
+var API = 'https://clouddebugger.googleapis.com/v2/debugger';
+
+/**
+ * @constructor
+ */
+function Debugger() {
+  common.ServiceObject.call(this, {
+    parent: new Debug({}),
```

Comment:
Should accept the Debug client as an parameter instead of creating from scratch.